File: CS/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.Common.DataModel;
using Scaffolding.ValidationErrors.Common.DataModel.EntityFramework;
using Scaffolding.ValidationErrors.Model;

namespace Scaffolding.ValidationErrors.DepartmentContextDataModel {
    public class DepartmentContextDesignTimeUnitOfWork : DesignTimeUnitOfWork, IDepartmentContextUnitOfWork {
        static DesignTimeDepartmentRepository departmentsRepository = new DesignTimeDepartmentRepository();

        public DepartmentContextDesignTimeUnitOfWork() {
        }

        bool IDepartmentContextUnitOfWork.HasChanges() {
            return false;
        }
        IDepartmentRepository IDepartmentContextUnitOfWork.Departments {
            get { return departmentsRepository; }
        }
        public void Detach(object entity) {
        }
    }
}

File: VB/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentContextUnitOfWork.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.ValidationErrors.Common.Utils
Imports Scaffolding.ValidationErrors.Common.DataModel
Imports Scaffolding.ValidationErrors.Common.DataModel.EntityFramework
Imports Scaffolding.ValidationErrors.Model

Namespace Scaffolding.ValidationErrors.DepartmentContextDataModel
	Public Class DepartmentContextDesignTimeUnitOfWork
		Inherits DesignTimeUnitOfWork
		Implements IDepartmentContextUnitOfWork
		Private Shared departmentsRepository As New DesignTimeDepartmentRepository()

		Public Sub New()
		End Sub

		Private Function HasChanges() As Boolean Implements IDepartmentContextUnitOfWork.HasChanges
			Return False
		End Function
		Private ReadOnly Property Departments() As IDepartmentRepository Implements IDepartmentContextUnitOfWork.Departments
			Get
				Return departmentsRepository
			End Get
		End Property
		Public Overloads Sub Detach(ByVal entity As Object) Implements Scaffolding.ValidationErrors.Common.DataModel.IUnitOfWork.Detach
		End Sub
	End Class
End Namespace
